### DISCUSSION TITLE
Create a Supabase Design System

### DISCUSSION THREAD

**Discussion by kiwicopple** [supabase/supabase#195] [General]
This is a Hacktoberfest task! 
You can read more about Hacktoberfest [on our blog](https://supabase.io/blog/2020/09/11/supabase-hacktoberfest-2020) and see all tasks on our [Hacktoberfest Project Board](https://github.com/orgs/supabase/projects/5).

# Summary

[Design systems](https://uxdesign.cc/what-the-heck-is-a-design-system-c89a8ea73b0d) guide how a product looks and operates. As our product starts to grow, we need a structured way to think about our designs (preferably using Figma)

![image](https://user-images.githubusercontent.com/10214025/93012210-d6903880-f5d0-11ea-8a0b-c28a9145c753.png)

## Details

Some additional context about how Supabase operates:

- Dark mode first, light mode second.
- Everything must be simple and uncluttered
- Operations for power-users are hidden away behind tooltips and key commands
- Animations are used sparingly, only to delight
- We're not wedded to our font-choices, feel free to suggest new ones

**Comment by Rihanat** [comment]
@kiwicopple can I start this?

**Comment by kiwicopple** [comment]
Sure thing @Rihanat ! Would you use Figma? How do you plan to go about this one?

**Comment by Rihanat** [comment]
@kiwicopple i use Figma for my designs. Should I submit the png file or the link to the design is also needed.

**Comment by kiwicopple** [comment]
Might be nice to have a link eventually, but you can just start with a PNG :)

**Comment by Rihanat** [comment]
@kiwicopple I just make a pull request, kindly check. Thank you

**Comment by kiwicopple** [comment]
Thanks @Rihanat ! 

There are [new rules](https://hacktoberfest.digitalocean.com/hacktoberfest-update) for hacktoberfest, so I'm going to label your PR `hacktoberfest-accepted` and then leave this issue so that everyone else can have a go if they want. There's nothing more you need to do - your PR will count towards Hacktoberfest :)

To make it easier for people, I'll drop your design system here:

![Supabase design system](https://user-images.githubusercontent.com/10214025/94988281-a835c580-059e-11eb-8ee7-40597c1a64ec.png)

**Comment by Rihanat** [comment]
Thanks @kiwicopple, I’m new to opensource contribution perhaps this is my first time of contributing to opensource. Does that mean my design has been approved and will be counted as one of the project that I’ve done on on Hacktoberfest. Thank you for your response.

**Comment by kiwicopple** [comment]
If this is your first time contributing to OSS, you did an awesome job getting the PR set up @Rihanat 

> Does that mean my design has been approved and will be counted as one of the project that I’ve done on on Hacktoberfest.

Yes, this counts towards Hacktoberfest! You need to create 4 PR's, so you will have 3 more to go.

We will leave the issue open for others to attempt it, just like you did - there is nothing more that you need to do for this one

**Comment by kiwicopple** [comment]
Now that Hacktoberfest is over I'll move this over to Discussions 💬

**Comment by MildTomato** [comment]
We've now been building out a design system recently in figma internally.
I would love to share this with everyone soon 🚀
